update_translations.py

```diff
@@ -0,0 +1,116 @@
+import subprocess
+import os
+import deploy_config as conf
+import xml.etree.ElementTree as ET
+
+# name of the master hq translations file to be updated
+hq_translations_filename = 'messages_en-2.txt'
+
+# relative path to the subfolder within the javarosa repo containing the messages_default.txt file
+javarosa_subfolder = './j2me/shared-resources/resources'
+
+# relative path to the subfolder within the commcare repo containing the messages_cc_default.txt file
+commcare_subfolder = './application/resources'
+
+# relative path to the subfolder within the commcare-odk repo containing the messages_ccodk_default.txt file
+ccodk_messages_subfolder = './app/assets/locales'
+
+# relative path to the subfolder within the commcare-odk repo containing the strings.xml file
+ccodk_strings_subfolder = './app/res/values'
+
+javarosa_repo = 'javarosa'
+commcare_repo = 'commcare'
+commcare_odk_repo = 'commcare-odk'
+translations_repo = 'commcare-translations'
+
+javarosa_filename = 'messages_default.txt'
+commcare_filename = 'messages_cc_default.txt'
+ccodk_messages_filename = 'messages_ccodk_default.txt'
+ccodk_strings_filename = 'strings.xml'
+
+javarosa_header_string = '# *** messages_default.txt ***'
+commcare_header_string = '# *** messages_cc_default.txt ***'
+ccodk_messages_header_string = '# *** messages_ccodk_default.txt ***'
+ccodk_strings_header_string = '# *** strings.xml ***'
+
+header_strings = [javarosa_header_string, commcare_header_string, ccodk_messages_header_string, ccodk_strings_header_string]
+
+strings_namespace = '{http://strings_namespace}'
+
+
+def update_translations(new_version_number):
+    new_javarosa_text = get_updated_translations(javarosa_repo, javarosa_subfolder, javarosa_filename)
```

Comment:
Before starting any real processing it would be good to check that the `translations_repo` doesn't have any pending changes. Something similar to https://github.com/dimagi/mobile-deploy/blob/master/deploy_git.py#L25-L31 should work. This will prevent the script from crashing halfway through with `git checkout -b foo` fails due to pending changes.